pkg/oc/admin/diagnostics/diagnostics/systemd/analyze_logs.go

```diff
@@ -99,8 +99,13 @@ func (d AnalyzeLogs) Check() types.DiagnosticResult {
 					if err := json.Unmarshal(bytes, &entry); err != nil {
 						r.Debug("DS0003", fmt.Sprintf("Couldn't read the JSON for this log message:\n%s\nGot error %s", string(bytes), errStr(err)))
 					} else {
-						if lineCount > 500 && stampTooOld(entry.TimeStamp, timeLimit) {
-							r.Debug("DS0004", fmt.Sprintf("Stopped reading %s log: timestamp %s too old", unitName, entry.TimeStamp))
+						old, unixTimestamp, err := stampTooOld(entry.TimeStamp, timeLimit)
+						if err != nil {
+							r.Error("DS0004", err, fmt.Sprintf("Stopped reading %s log, timestamp %s is invalid", unitName, entry.TimeStamp))
+							break
```

Comment:
Subtle difference here. This will end the loop the first time it sees a malformed log line (if that ever happens! not sure I've ever seen it). Previously it would only end the loop if there was an error *and* it had seen more than 500 lines.

It would make more sense to me to skip the log line with a `continue` here and keep looking.